tests/tests_fabric/plugins/environments/test_xla.py

```diff
@@ -44,21 +44,25 @@ def test_default_attributes(*_):
 
 
 @RunIf(tpu=True)
-@mock.patch.dict(
-    os.environ,
-    {
-        **os.environ,
-        "TPU_MESH_CONTROLLER_ADDRESS": "1.2.3.4",
-        "TPU_MESH_CONTROLLER_PORT": "500",
-        "XRT_SHARD_WORLD_SIZE": "1",
-        "XRT_SHARD_ORDINAL": "0",
-        "XRT_SHARD_LOCAL_ORDINAL": "2",
-        "XRT_HOST_ORDINAL": "3",
-    },
-    clear=True,
-)
+@mock.patch.dict(os.environ, os.environ.copy(), clear=True)
 def test_attributes_from_environment_variables():
     """Test that the default cluster environment takes the attributes from the environment variables."""
+    from torch_xla.experimental import pjrt
+
+    if pjrt.using_pjrt():
+        # FIXME
+        envvars = {}
```

Comment:
I believe the `XLAEnvironment` needs an update, particularly these

https://github.com/Lightning-AI/lightning/blob/da883e73b941112aac7a538d71def6af2e83113b/src/lightning/fabric/plugins/environments/xla.py#L40-L50
https://github.com/Lightning-AI/lightning/blob/da883e73b941112aac7a538d71def6af2e83113b/src/lightning/fabric/plugins/environments/xla.py#L77-L81

For reference, this is the list of envvars in the v4-8 machine:

```python
{
    "PL_RUN_TPU_TESTS": "1",
    "SHELL": "/bin/bash",
    "GLIBCXX_FORCE_NEW": "1",
    "PWD": "/home/runner/lightning/tests/tests_pytorch",
    "LOGNAME": "runner",
    "XDG_SESSION_TYPE": "tty",
    "GLIBCPP_FORCE_NEW": "1",
    "MOTD_SHOWN": "pam",
    "LD_PRELOAD": "/usr/lib/x86_64-linux-gnu/libtcmalloc.so.4",
    "HOME": "/home/runner",
    "LANG": "C.UTF-8",
    "SSH_CONNECTION": "13.87.244.202 36864 10.130.0.10 22",
    "CLOUDSDK_PYTHON": "/usr/bin/python3",
    "XDG_SESSION_CLASS": "user",
    "USER": "runner",
    "SHLVL": "2",
    "PJRT_DEVICE": "TPU",
    "XDG_SESSION_ID": "3",
    "LD_LIBRARY_PATH": ":/usr/local/lib",
    "XDG_RUNTIME_DIR": "/run/user/2008",
    "SSH_CLIENT": "13.87.244.202 36864 22",
    "PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin",
    "DBUS_SESSION_BUS_ADDRESS": "unix:path=/run/user/2008/bus",
    "ALLOW_MULTIPLE_LIBTPU_LOAD": "1",
    "OLDPWD": "/home/runner/lightning",
    "_": "/usr/bin/python3",
    "COVERAGE_RUN": "true",
    "PYTHONPATH": "/home/runner/lightning/tests",
    "PYTORCH_NVML_BASED_CUDA_CHECK": "1",
    "TF_CPP_MIN_LOG_LEVEL": "1",
    "GRPC_VERBOSITY": "ERROR",
    "TPU_ML_PLATFORM": "PyTorch/XLA",
    "TF_GRPC_DEFAULT_OPTIONS": "grpc.keepalive_time_ms=60000,grpc.keepalive_timeout_ms=14400000,grpc.http2.max_pings_without_data=0,grpc.http2.min_ping_interval_without_data_ms=300000",
    "XLA_FLAGS": " --xla_cpu_enable_fast_math=false --xla_gpu_simplify_all_fp_conversions=false --xla_gpu_force_compilation_parallelism=8",
    "TPU_LIBRARY_PATH": "/usr/local/lib/python3.8/dist-packages/torch_xla/lib/libtpu.so",
    "PJRT_LOCAL_PROCESS_RANK": "0",
    "PJRT_LOCAL_PROCESS_COUNT": "4",
    "TPU_CHIPS_PER_PROCESS_BOUNDS": "1,1,1",
    "TPU_PROCESS_BOUNDS": "2,2,1",
    "CLOUD_TPU_TASK_ID": "0",
    "TPU_PROCESS_ADDRESSES": "localhost:8476,localhost:8477,localhost:8478,localhost:8479",
    "TPU_VISIBLE_CHIPS": "0",
    "TPU_PROCESS_PORT": "8476",
}
```

Reply:
The xla env probably needs to check too whether pjrt is used. They added new env vars here: 
https://github.com/pytorch/xla/blob/master/torch_xla/core/xla_env_vars.py

As main_address, should it pick TPU_PROCESS_ADDRESSES[0] and TPU_PROCESS_PORT for the port?

Reply:
@will-cromar your help may needed here, you know more history about these env vars. Does `TPU_PROCESS_ADDRESSES` or `TPU_PROCESS_PORT` apply to PJRT?

Reply:
Yes. These are used by libtpu for initialization. PJRT sets these variables in `xmp.spawn` to configure multiprocessing: https://github.com/pytorch/xla/blob/e5b7cb45b010357d6c58bae2035817c671615d38/torch_xla/experimental/tpu.py#L181-L231

Reply:
@awaelchli what does Lightning use the `main_address`, `main_port`, and `node_rank ` in the `XLAEnvironment` for? I remember I tried to look for their usage, but did not find a solid answer from the code, so I left them as is. If you give me some info on their usage, I can update them.

Reply:
For torch ddp, the main_address and main_port are used for `torch.distributed.init_process_group`. Since XLA does this kind of equivalent thing in xmp.spawn, and if there is no real usage of these methods anywhere else, then you may want to define them by raising NotImplementedError.

Reply:
Yes, I agree. It make sense to raise `NotImplementedError` in this case. I updated `main_address` and `main_port`. Not so sure about `node_rank`, it seems to be used by `_DataConnector`